ocf_data_sampler/load/nwp/providers/icon.py

```diff
@@ -0,0 +1,101 @@
+"""DWD ICON Loading"""
+
+import pandas as pd
+import xarray as xr
+import fsspec
+
+from ocf_data_sampler.load.nwp.providers.utils import open_zarr_paths
+
+def transform_to_channels(nwp : xr.Dataset):
```

Comment:
Am I right in thinking that the input here is an xarray `Dataset` which has multiple data variables for each NWP variable and we want to go from that to a `DataArray` (e.g. one data variable but an extra channel dimension?)

I think a simpler approach might be to do something like what is done here https://github.com/openclimatefix/ocf_datapipes/blob/main/ocf_datapipes/load/nwp/providers/gfs.py#L26 where we use `to_array()` on the Dataset to convert it to a `DataArray` and then rename the `variable` dimension which is created with `to_array()` to `channel`

But I may have misunderstood the intention/need for this function

Reply:
you are perfectly right, I deleted this and use to_array() instead, thx for pointing it out!